src/backend/controllers/pages.ts

```diff
@@ -62,6 +62,47 @@ class Pages {
     return nullFilteredPages;
   }
 
+  /**
+   * Group given pages with descending order of creation time
+   *
+   * @param {pages[]} pages - pages to group
+   * @returns {Page[]}
+   */
+  public static group(pages: Page[]): Page[] {
+    const result: Page[] = [];
+    const obj:Record<string, Array<Page>> = {};
+
+    pages.sort((a, b) => {
+      if (a.body.time > b.body.time) {
+        return 1;
+      }
+
+      if (a.body.time < b.body.time) {
+        return -1;
+      }
+
+      return 0;
+    }).forEach(doc => {
+      if (doc._parent === '0' && typeof doc._id === 'string') {
```

Comment:
move this check to a separate statement (like `isRootPage(page)`). And why do we need the checking?

Reply:
Fixed. Because the behavior is different between root page and non-root page if the page is root then first set key with its `_id` and then assign itself to the key just created, On the other hand if page is non-root then it adds key with its `_id` just like root page but if there key same as its `_parent` then it should be assigned into that key.